sci-electronics/kicad/kicad-4.0.6.ebuild

```diff
@@ -0,0 +1,191 @@
+for lang in ${LANGS} ; do
+	IUSE="${IUSE} l10n_${lang}"
+done
+unset lang
+
+REQUIRED_USE="
+	python? ( ${PYTHON_REQUIRED_USE} )"
+
+COMMON_DEPEND=">=x11-libs/wxGTK-3.0.2:${WX_GTK_VER}[X,opengl]
+	python? (
+		dev-python/wxpython:${WX_GTK_VER}[opengl,${PYTHON_USEDEP}]
+		${PYTHON_DEPS}
+	)
+	>=dev-libs/boost-1.61[context,nls,threads,python?,${PYTHON_USEDEP}]
+	github? (
+		libressl? ( dev-libs/libressl:0= )
+		!libressl? ( dev-libs/openssl:0= )
+	)
+	media-libs/glew:0=
+	media-libs/freeglut
+	media-libs/mesa
+	sys-libs/zlib
+	x11-libs/cairo"
+DEPEND="${COMMON_DEPEND}
+	doc? ( app-doc/doxygen )
+	i18n? ( virtual/libintl )
+	python? ( dev-lang/swig:0 )"
+RDEPEND="${COMMON_DEPEND}
+	sci-electronics/electronics-menu"
+
+pkg_setup() {
+	use python && python-single-r1_pkg_setup
+	setup-wxwidgets
+}
+
+src_prepare() {
+	xdg_src_prepare
+
+	# Patch to work with >=boost 1.61
+	eapply "${FILESDIR}/${PN}-boost-1.61.patch"
+
+	# remove all the non unix file endings
+	while IFS="" read -d $'\0' -r f; do
+		edos2unix "${f}"
+	done < <(find "${S}" -type f -name "*.desktop" -print0)
+
+	# Remove cvpcb desktop file while it does nothing
+	rm "${WORKDIR}/${P}/resources/linux/mime/applications/cvpcb.desktop" || die
+
+	# Handle optional minimal install.
+	if use minimal; then
+		# remove templates as they are not needed to run binaries
+		sed -e '/add_subdirectory( template )/d' -i CMakeLists.txt || die
+	else
+		# create a link to the parts library in the main project folder
+		ln -s "${WORKDIR}/kicad-library-${PV}" "${S}/${PN}-library" || die
+		# create a link to the footprints library and add cmake build rule for it
+		ln -s "${WORKDIR}/${PN}-footprints-${PV}" "${S}/${PN}-footprints" || die
+		cp "${FILESDIR}/${PN}-footprints-cmakelists.txt" "${WORKDIR}/${PN}-footprints-${PV}/CMakeLists.txt" || die
+		# add the libraries directory to cmake as a subproject to build
+		sed "/add_subdirectory( bitmaps_png )/a add_subdirectory( ${PN}-library )" -i CMakeLists.txt || die
+		# add the footprints directory to cmake as a subproject to build
+		sed "/add_subdirectory( ${PN}-library )/a add_subdirectory( ${PN}-footprints )" -i CMakeLists.txt || die
+		# remove duplicate uninstall directions for the library module
+		sed '/make uninstall/,/# /d' -i ${PN}-library/CMakeLists.txt || die
+	fi
+
+	# Add internationalization for the GUI
+	if use i18n; then
+		# create a link to the translations library in the main project folder
+		ln -s "${WORKDIR}/${P}-i18n" "${S}/${PN}-i18n" || die
+		# Remove unused languages. Project generates only languages specified in the
+		# file in LINGUAS in the subproject folder. By default all languages are added
+		# so we sed out the unused ones based on the user l10n_* settings.
+		local lang
+		for lang in ${LANGS}; do
+			if ! use l10n_${lang}; then
+				lang="${lang//-/_}"  # Needed to turn zh-CN to zh_CN as KiCad does not follow l10n standard here
+				sed "/${lang}/d" -i ${PN}-i18n/LINGUAS || die
```

Comment:
Honestly, I don't like this sed. It's easy to make a mistake in LANGS list, and it will never fail. Maybe add a verification grep before it, so that it would fail if lang is not in the file? But that's something to be though about later.